sources/packages/backend/apps/api/src/route-controllers/models/pagination.dto.ts

```diff
@@ -24,9 +24,10 @@ abstract class PaginationOptionsAPIInDTO {
   page: number;
   /**
    * Page size or records per page.
+   * -1 means all records.
```

Comment:
Min could be also 0 here as 0 is returning all records.

![image](https://user-images.githubusercontent.com/54600590/221296247-35a6299c-03e3-4f78-81e2-b4d77711d51c.png)

Reply:
For the data tables loading data from the server we should not allow the request for all the data. Is it possible in the component? I am asking because the "-1 means all records.".

Reply:
In the example here you are showing the `v-pagination` not the default one. below is the default one 
![image](https://user-images.githubusercontent.com/77353155/221298564-4f1dcd78-fe54-451b-a018-ab78767ad334.png)

Reply:
According to what @andrewsignori-aot which is right min value must be 1 right?

Reply:
agree with @andrewsignori-aot. min is 0 now, so the API will restrict when all records are fetched. As discussed I could find a way to disable `All` or to pass pagination options as primevue, the only option currently is to extend the footer and add a custom one. Leaving it like this and When we decide to change all data tables, if there isn`t any option available we can invest more time on it as discussed